README.md

```diff
@@ -157,17 +157,14 @@ changes reflected in your python environment.
 ### Develop Mode
 
 If you'd like to build rustworkx in debug mode and use an interactive debugger
-while working on a change you can use `python setup.py develop` to build
-and install rustworkx in develop mode. This will build rustworkx without
-optimizations and include debuginfo which can be handy for debugging. Do note
-that installing rustworkx this way will be significantly slower then using
-`pip install` and should only be used for debugging/development.
+while working on a change you can set `SETUPTOOLS_RUST_CARGO_PROFILE="dev"`
```

Comment:
Do we want to set a minimum version for setuptools-rust if we document this. Historically this option didn't exist when rustworkx started. Looking at the changelog it was less recent than I thought but still relatively recent in 1.8.0 from Oct. 2023.

Reply:
I will update the README to document that.

In my view, any software newer than our `rustc` should be fair to ask for people building from source. The same applies for `setuptools` and `pip`.

Reply:
Done, let's say the version from February 2024 is the minimum one.

I also fixed a place where we list an incorrect rustc version.